src/api/lessons.ts

```diff
@@ -0,0 +1,28 @@
+import axios, { AxiosResponse } from 'axios';
+import Lesson from '@/models/lesson/lesson';
+import buildUrl from './buildUrl';
+
+async function getLessonById(lessonId: number): Promise<Lesson> {
+  try {
+    const response: AxiosResponse = await axios.get(buildUrl(`/lesson/${lessonId}`));
+
+    const lesson = new Lesson(
+      response.data.data[0].lessonId,
```

Comment:
@osamaramihafez we should address this at some point and not return an array for a single entity endpoint

Reply:
Agreed, lots to clean up tbh. Let's create tickets for issues as they arise. It'll be easier to keep track in the backlog insha'Allah

Reply:
Where should we be creating these tickets?

Reply:
Created the issues in the backend repo